### PR TITLE
Added support for Template and Auto Layout Options

### PR DESCRIPTION
**Pull request by carlos-kelly**
Adds support for AutoLayout prop plus Template File injection. Also updates default template.

### MDX + Custom Template File
Screen Shot 2019-11-02 at 6 38 08 PM

### MDX + Default Template
Screen Shot 2019-11-02 at 6 38 27 PM

### MD + Default Template
Screen Shot 2019-11-02 at 6 38 43 PM

### MD + Custom Template
Screen Shot 2019-11-02 at 6 39 03 PM

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -5,9 +5,25 @@ const { parse } = require('./args');
...
 const main = async () => {
-  const { action, port, title, srcFilePath, themeFilePath } = await parse();
+  const {
+    action,
+    port,
+    title,
+    srcFilePath,
+    themeFilePath,
+    templateFilePath,
+
```
Comment (rgerstenberger) [inline bin/cli.js]: Why the newline here?

Comment (carlos-kelly) [reply bin/cli.js]: Prettier? Let me fix lol

File: bin/cli.js [hunk 2]
```diff
@@ -5,9 +5,24 @@ const { parse } = require('./args');
...
+  } = await parse();
   const run = actions[action];
-  await run({ port, title, srcFilePath, themeFilePath });
+  await run({
+    port,
+    title,
+    srcFilePath,
+    themeFilePath,
+    autoLayout,
+    templateFilePath
```
Comment (ryan-roemer) [inline bin/cli.js]: Very minor. Do we want to do the same order as the above object destructuring?

File: lib/actions.js [hunk 3]
```diff
@@ -34,12 +34,26 @@ import theme from '${themeFilePath}';
...
+const TEMPLATE_RE = /\/\/ SPECTACLE_CLI_TEMPLATE_START[\s\S]*?\/\/ SPECTACLE_CLI_TEMPLATE_END/gm;
+const templateTmpl = templateFilePath => `
+// SPECTACLE_CLI_TEMPLATE_START
+import template from '${templateFilePath}';
+// SPECTACLE_CLI_TEMPLATE_END
+`;
+
+const AUTOLAYOUT_RE = /\/\/ SPECTACLE_CLI_AUTOLAYOUT_START[\s\S]*?\/\/ SPECTACLE_CLI_AUTOLAYOUT_END/gm;
+const autolayoutTmpl = autolayout => `
+const autoLayout = ${Boolean(autolayout)};
```
Comment (ryan-roemer) [inline lib/actions.js]: Let’s also add the comment markers like the other examples do

**Pull request merged**